vpr/src/place/net_cost_handler.h

```diff
@@ -190,17 +190,45 @@ class NetCostHandler {
     vtr::vector<ClusterNetId, double> proposed_net_cost_;
     vtr::vector<ClusterNetId, NetUpdateState> bb_update_status_;
 
+    /**
+     * @brief This class is used to store the inverse of average channel
```

Comment:
I am not sure. The `DimRange(size_t begin, size_t end)` constructor gets size_t arguments, meaning that we can't start a range from -1.

Reply:
All `operator[]`s implemented in vtr_ndoffsetmatrix.h have an argument of type `size_t`. The main purpose of this calss is overriding `operator[]` with an int argument and hiding the one with size_t argument.

If we want to remove this class, `vtr::NdOffsetMatrix`'s code need to be changed to accept int indices instead of size_t indices.

Reply:
That makes sense to me ... makes a more general utility class.